traffic_ops/traffic_ops_golang/physlocation/phys_locations.go

```diff
@@ -215,3 +219,307 @@ func deleteQuery() string {
+		api.HandleErr(w, r, tx, code, usrErr, sysErr)
+		return
+	}
+	alerts := tc.CreateAlerts(tc.SuccessLevel, "physLocation was created.")
+	w.Header().Set("Location", fmt.Sprintf("/api/%d.%d/phys_locations?name=%s", inf.Version.Major, inf.Version.Minor, physLocation.Name))
+	api.WriteAlertsObj(w, r, http.StatusCreated, alerts, physLocation)
+	return
+}
+
+func UpdatePhysLocation(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	tx := inf.Tx.Tx
+	physLocation, readValErr := readAndValidateJsonStruct(r)
+	if readValErr != nil {
+		api.HandleErr(w, r, tx, http.StatusBadRequest, readValErr, nil)
+		return
+	}
+
+	requestedID := inf.Params["id"]
+
+	intRequestId, convErr := strconv.Atoi(requestedID)
+	if convErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, fmt.Errorf("physLocation update error: %w, while converting from string to int", convErr), nil)
+	}
+	// check if the entity was already updated
+	userErr, sysErr, errCode = api.CheckIfUnModified(r.Header, inf.Tx, intRequestId, "phys_location")
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		return
+	}
+
+	//update name and description of a phys_location
+	query := `
+	UPDATE phys_location pl
+	SET
+		address = $1,
+		city = $2,
+		comments = $3,
+		email = $4,
+		name = $5,
+		phone = $6,
+		poc = $7,
+		region = $8,
+		short_name = $9,
+		state = $10,
+		zip = $11
+	WHERE
+		pl.id = $12
+	RETURNING
+		pl.id, pl.last_updated
+`
+
+	err := tx.QueryRow(
+		query,
+		physLocation.Address,
+		physLocation.City,
+		physLocation.Comments,
+		physLocation.Email,
+		physLocation.Name,
+		physLocation.Phone,
+		physLocation.POC,
+		physLocation.RegionID,
+		physLocation.ShortName,
+		physLocation.State,
+		physLocation.Zip,
+		requestedID,
+	).Scan(
+		&physLocation.ID,
+		&physLocation.LastUpdated,
+	)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			api.HandleErr(w, r, tx, http.StatusBadRequest, fmt.Errorf("physLocation with ID: %v not found", physLocation.ID), nil)
+			return
+		}
+		usrErr, sysErr, code := api.ParseDBError(err)
+		api.HandleErr(w, r, tx, code, usrErr, sysErr)
+		return
+	}
+	alerts := tc.CreateAlerts(tc.SuccessLevel, "physLocation was updated")
+	api.WriteAlertsObj(w, r, http.StatusOK, alerts, physLocation)
+	return
+}
+
+func DeletePhysLocation(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	tx := inf.Tx.Tx
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	id := inf.Params["id"]
+	exists, err := dbhelpers.PhysLocationExists(tx, id)
+	if err != nil {
+		api.HandleErr(w, r, tx, http.StatusInternalServerError, nil, err)
+		return
+	}
+	if !exists {
+		if id != "" {
```

Comment:
Should we move the `id != ""` check happen right after we get id grom `inf.Params`?

Reply:
Fixed it.